packages/react/src/components/DataTable/Table.tsx

```diff
@@ -66,13 +74,69 @@ export const Table = ({
     [`${prefix}--data-table--sticky-header`]: stickyHeader,
     [`${prefix}--data-table--visible-overflow-menu`]: !overflowMenuOnHover,
   });
+
+  const setTableAlignment = useCallback(() => {
+    const fragment = document.createDocumentFragment();
+    const canvas = document.createElement('canvas');
+    fragment.appendChild(canvas);
+    const context = canvas.getContext('2d');
+
+    if (tableRef.current && context) {
+      const isMultiline = Array.from(
+        tableRef.current.querySelectorAll('td')
+      ).some((td) => {
```

Comment:
yeah I agree it's not the most elegant code out there 😅. I couldn't figure out any other way to do it though, any one cell could be wrapping content, so we need to check all of them - or stop when we find one that is wrapping, which I'm doing but if none of them are wrapping then yeah, it'll run through all-.
I can add a large datatable to test, how many rows/columns are you thinking will be a good enough test?

Reply:
1000+ rows is where we've seen other performance issues https://github.com/carbon-design-system/carbon/issues/4338

If we're not 100% confident in the performance, we could consider putting the behavior behind a an experimental prop so users can opt into it rather than it be on by default. Document that it has performance implications, etc